src/protocol/sort/shuffle.rs

```diff
@@ -0,0 +1,99 @@
+use crate::helpers::prss::PrssSpace;
+
+#[allow(dead_code)]
+
+pub struct Shuffle {}
+
+impl Shuffle {
+    /// The Fisher–Yates shuffle is an algorithm for generating a random permutation of a finite
+    /// sequence—in plain terms, the algorithm shuffles the sequence. The algorithm effectively
+    /// puts all the elements into a hat; it continually determines the next element by randomly
+    /// drawing an element from the hat until no elements remain. The algorithm produces an unbiased
+    /// permutation: every permutation is equally likely. This algorithm takes time proportional to
+    /// the number of items being shuffled and shuffles them in place.
+    #[allow(dead_code)]
+    fn generate_random_sequence(
+        batchsize: usize,
+        random_value: u128,
+        left_permute: bool,
+        prss: &PrssSpace,
+    ) -> Vec<usize> {
```

Comment:
I am pretty sure we can make it efficient by returning an iterator instead as there is no input array that must be shuffled, we basically generate a random permutation of (1, N)

Reply:
I have replaced everything with a call to shuffle crate which returns the permutation. However, that makes it not return an iterator but an array of permutation.